src/contexts/ReducerFunc.js

```diff
@@ -38,9 +46,45 @@ export const productReducer = (state, action) => {
                 bySamsung: null,
                 byRating: null,
                 bySearch: "",
-                byRange: null
+                byRange: null,
+                byCategory: []
             }
         default:
             return state;
     }
-}
\ No newline at end of file
+}
+
+
+const addCartReducer = (state, action) => {
+    switch (action.type) {
+        case "ADD_TO_CART":
+            return {addCart: [...state.addCart, { ...action.payload, qty: 1 }] };
+        case "REMOVE_FROM_CART":
+            return {addCart:state.addCart.filter((i) => i._id !== action.payload._id)}
+        case "INCREMENT" :
+            const increaseValue = state.addCart.map(data => data.id === action.payload.id ?
+                 {...action.payload,qty:action.payload.qty+1} : data);
+            return{addCart:increaseValue};
+        case "DECREMENT" :
+            if (action.payload.qty>1){
+                const dicreaseValue = state.addCart.map(data => data.id === action.payload.id ?
+                        {...action.payload,qty:action.payload.qty-1} : data);
+                    return{addCart:dicreaseValue};
```

Comment:
you can combine increment and decrement into one function along with the action type as well, like ```UDATE_CART_QTY```